File: Go/chi/SerializationHandlers.go
package main

import (
	"encoding/json"
	"log"
	"net/http"
)

type StaticUser struct {
	Id       int      `json:"id"`
	Username string   `json:"username"`
	Email    string   `json:"email"`
	IsActive bool     `json:"is_active"`
	Roles    []string `json:"roles"`
}

func HandleUserSerialization(w http.ResponseWriter, r *http.Request) {
	user, err := json.Marshal(StaticUser{
		Id:       1,
		Username: "JohnDoe",
		Email:    "johndoe@gmail.com",
		IsActive: true,
		Roles:    []string{"user", "admin"},
	})
	if err != nil {
		log.Println(err)
		w.WriteHeader(500)
		return
	}
	w.Header().Set("Content-Type", "application/json")
	w.Write(user)
}

File: Python/Litestar/serialization_handlers.py
from litestar import get
from pydantic import BaseModel


class StaticUser(BaseModel):
    id: int
    username: str
    email: str
    is_active: bool
    roles: list[str]

@get("/user/json")
async def handle_user_serialization() -> StaticUser:
    return StaticUser(
        id = 1,
        username = "JohnDoe",
        email = "johndoe@gmail.com",
        is_active = True,
        roles = ["user", "admin"],
    )